pyiron_base/project/generic.py

```diff
@@ -1548,6 +1548,32 @@ def __repr__(self):
             {"groups": self.list_dirs(skip_hdf5=True), "nodes": self.list_nodes()}
         )
 
+    def __getstate__(self):
+        return {
+            "user": self.user,
+            "sql_query": self.sql_query,
+            "history": self._history,
+            "root_path": self._root_path,
+            "project_path": self._project_path,
+            "filter": self._filter,
+            "inspect_mode": self._inspect_mode,
+        }
+
+    def __setstate__(self, state):
+        self.user = state["user"]
+        self.sql_query = state["sql_query"]
+        self._history = state["history"]
+        self._root_path = state["root_path"]
+        self._project_path = state["project_path"]
```

Comment:
Ideally this would be set in `GenericPath/ProjectPath.__setstate__`, but I'm not sure why even have those as separate classes.

Reply:
I implemented `__getstate__()` and `__setstate__()` in `GenericPath` and `ProjectPath`. Still I agree that the interface should be refactored.